lib/reline/io/windows.rb

```diff
@@ -115,7 +115,7 @@ def initialize(dllname, func, import, export = "0", calltype = :stdcall)
       def call(*args)
         import = @proto.split("")
         args.each_with_index do |x, i|
-          args[i], = [x == 0 ? nil : +x].pack("p").unpack(POINTER_TYPE) if import[i] == "S"
+          args[i], = [x == 0 ? nil : x&.+@].pack("p").unpack(POINTER_TYPE) if import[i] == "S"
```

Comment:
The current patch is the result of a minimization of fixes because I could not figure out the intent of args repacking.
By checking the code around ruby 1.9 to 2.0, I found that DL::CFunc of Ruby/DL (Ruby/DL2), the backend of WIN32API at that time, accepts only integers.
I think the repacking itself is unnecessary because the current Fiddle allows more flexible passing of arguments.
```
def call(*args)
  return @func.call(*args)
end
```
The above may be enough. Is there something I'm missing?

Reply:
I think the NULL problem is fixed in 9b3ffd8
String `0.chr * n` `+"\0\0\0..."` seems to be always passed to import[i]=="S" (import is 'P') argument except
`call_with_console_handle(@ScrollConsoleScreenBuffer, scroll_rectangle, nil, destination_origin, fill)`
and this line is removed in 9b3ffd8

Reply:
Thanks for the suggestion. I was not aware of 9b3ffd8 at all.
However, I am worried about touching a code that I don't know the basis for, and I now think that code is unnecessary.
So I think it was done in https://github.com/ruby/reline/pull/775/commits/060ba140ed431d368ab322746c12acd175e61c7b .

Reply:
```ruby
def call(*args)
  return @func.call(*args)
end
```
This looks good to me.
However, I don't have confidence that this works in all supported ruby versions and fiddle version. I think it is better to avoid risk since next ruby release is close in time.
How about leaving it as is (as original `[x == 0 ? nil : +x].pack("p")....`) for now, and change it after adding windows ci?

Reply:
I understand your point of consideration.
I also know that there is no specific problem that the patch will solve.
reverted in eb5959c11c0fdd935658457a2398a0f4a48bde8a 5443ab5225e1d046bf37d23e459d6ac71e17d5aa